File: 003.py
def factorise(factorise):
    i = 2
    while i <= factorise:
        if factorise % i == 0:
            factorise = factorise / i;
            print(i)
        if factorise % i != 0:
            i += 1
factorise(16)

File: 003.cc
#include <iostream>
#include <cstdlib>
#include <cmath>
using namespace std;

int main()
{
  long factorise = 600851475143;
  long largestfactor = 1;
  long i = 2;


  while (i <= factorise)
  {
    if (factorise % i == 0)
    {
      factorise = factorise / i;
      cout << i << endl;
    }
    if (factorise % i != 0)
    {
      i += 1;
    }
  }

}
